toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/LocalSceneViewProxy.kt

```diff
@@ -0,0 +1,169 @@
+ USA
+ 
+ email: contracts@esri.com
+ */
+
+package com.arcgismaps.toolkit.geoviewcompose
+
+import androidx.compose.runtime.Stable
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.view.AnimationCurve
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.LocalSceneView
+import com.arcgismaps.mapping.view.ScreenCoordinate
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlin.time.DurationUnit
+
+/**
+ * Used to perform operations on a composable [LocalSceneView].
+ *
+ * There should be a one-to-one relationship between a LocalSceneViewProxy and a composable
+ * [LocalSceneView]. This relationship is established by passing an instance of LocalSceneViewProxy
+ * to the composable [LocalSceneView] function. Operations can only be performed once the associated
+ * composable LocalSceneView has entered the composition. Operations performed when the associated
+ * composable LocalSceneView is not in the composition will fail gracefully, i.e. won't throw
+ * exceptions but won't return a successful result.
+ *
+ * @since 300.0.0
+ */
+@Stable
+public class LocalSceneViewProxy : GeoViewProxy("LocalSceneView") {
+    /**
+     * The view-based [com.arcgismaps.mapping.view.LocalSceneView] that this LocalSceneViewProxy
+     * will operate on. This should be initialized by the composable [LocalSceneView] when it enters
+     * the composition and set to null when it is disposed by calling [setLocalSceneView].
+     *
+     * @since 300.0.0
+     */
+    private var localSceneView: LocalSceneView? = null
+        set(value) {
+            setGeoView(value)
+            field = value
+        }
+
+    /**
+     * Sets the [localSceneView] parameter on this operator. This should be called by the composable
+     * [LocalSceneView] when it enters the composition and set to null when it is disposed.
+     *
+     * @since 300.0.0
+     */
+    internal fun setLocalSceneView(localSceneView: LocalSceneView?) {
+        this.localSceneView = localSceneView
+    }
+
+    private val nullSceneViewErrorMessage: String =
+        "LocalSceneView must be part of the composition when this member is called."
+
+    /**
+     * Pans or zooms the local scene view using animation to the specified viewpoint location. The
+     * animation takes place over the specified duration. The animation curve defines the animation
+     * easing function.
+     *
+     * @param viewpoint The viewpoint that should be set on the view.
+     * @param duration The amount of time to move to the new viewpoint.
+     * @param animationCurve The type of animation curve.
+     * @return a [Result] of true if the viewpoint was successfully set, false otherwise.
+     * @since 300.0.0
+     */
+    public suspend fun setViewpointAnimated(
+        viewpoint: Viewpoint,
+        duration: Duration = 3.seconds,
+        animationCurve: AnimationCurve
+    ): Result<Boolean> {
```

Comment:
The defaulted parameter needs to come last, otherwise a caller is forced to use named parameters when writing code like this:
```
setViewpointAnimated(
   myViewpoint,
   AnimationCurve.EaseInCirc
)
```

```suggestion
    public suspend fun setViewpointAnimated(
        viewpoint: Viewpoint,
        animationCurve: AnimationCurve,
        duration: Duration = 3.seconds
    ): Result<Boolean> {
```

Reply:
Fixed